### PR TITLE
Remove many unused variables in tests.

### PR DESCRIPTION
**Pull request by anntzer**
... as detected by enabling F841 in flake8.

A lot of these are instantiations of figures which are not used because
the test then uses the pyplot interface anyways; the figure doesn't need
to be explicitly instantiated as tests always run with no preexisting
figure (via an autouse fixture).

Split out from #13932 for reviewability.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/tests/test_axes.py [hunk 1]
```diff
@@ -1621,18 +1587,12 @@ def test_hist_steplog():
...
-    ax = plt.subplot(4, 1, 2)
-    plt.hist(data_pos, 100, histtype='stepfilled', log=True)
-
-    ax = plt.subplot(4, 1, 3)
-    plt.hist(data, 100, weights=weights, histtype='stepfilled', log=True)
-
-    ax = plt.subplot(4, 1, 4)
-    plt.hist(data_big, 100, histtype='stepfilled', log=True,
-             orientation='horizontal')
+    fig, axs = plt.subplots(4)
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_axes.py]: This puts an unused variable in?

File: lib/matplotlib/tests/test_axes.py [hunk 2]
```diff
@@ -5301,16 +5255,13 @@ def test_title_pad():
...
-    # set default title location
-    plt.rcParams['axes.titlelocation'] = 'center'
-
     ax.set_title('aardvark')
     ax.set_title('left', loc='left')
     ax.set_title('right', loc='right')
 
     assert 'left' == ax.get_title(loc='left')
     assert 'right' == ax.get_title(loc='right')
-    assert 'aardvark' == ax.get_title(loc='center')
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_axes.py]: Probably should keep both versions of this.

File: lib/matplotlib/tests/test_axes.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/tests/test_axes.py]: true, fixed

File: lib/matplotlib/tests/test_axes.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/tests/test_axes.py]: oops, that was removed by mistake.  restored.

File: lib/matplotlib/tests/test_mlab.py [hunk 3]
```diff
@@ -1835,7 +1828,7 @@ def test_specgram_angle(self):
                                      pad_to=self.pad_to_specgram,
                                      sides=self.sides,
                                      mode='angle')
-        specm = np.mean(spec, axis=1)
+        np.mean(spec, axis=1)
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_mlab.py]: That's doing nothing. `specs` is not modified and the return value is not used.

File: lib/matplotlib/tests/test_mlab.py [hunk 4]
```diff
@@ -1851,7 +1844,7 @@ def test_specgram_phase(self):
                                      pad_to=self.pad_to_specgram,
                                      sides=self.sides,
                                      mode='phase')
-        specm = np.mean(spec, axis=1)
+        np.mean(spec, axis=1)
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_mlab.py]: Doing nothing.

File: lib/matplotlib/tests/test_scale.py [hunk 5]
```diff
@@ -100,11 +100,11 @@ def test_logscale_transform_repr():
     # check that repr of log transform succeeds
     fig, ax = plt.subplots()
     ax.set_yscale('log')
-    s = repr(ax.transData)
+    repr(ax.transData)
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_scale.py]: Does nothing`? Or
```suggestion
    repr(ax.transData)  # smoke test that repr() succededs
```

**Comment by anntzer** [comment]
thanks, comments handled

**Pull request merged**